src/index.js

```diff
@@ -6,18 +6,70 @@ import "./styles.css";
 
 var elements = [];
 
-function newElement(type, x, y) {
+function newElement(type, x, y, width = 0, height = 0) {
   const element = {
     type: type,
     x: x,
     y: y,
-    width: 0,
-    height: 0,
+    width: width,
+    height: height,
     isSelected: false
   };
   return element;
 }
 
+function exportAsPNG({ background, visibleOnly, padding = 10 }) {
+  clearSelection();
+  drawScene();
+
+  let subCanvasX1 = Infinity;
+  let subCanvasX2 = 0;
+  let subCanvasY1 = Infinity;
+  let subCanvasY2 = 0;
+
+  elements.forEach(element => {
+    subCanvasX1 = Math.min(subCanvasX1, getElementAbsoluteX1(element));
+    subCanvasX2 = Math.max(subCanvasX2, getElementAbsoluteX2(element));
+    subCanvasY1 = Math.min(subCanvasY1, getElementAbsoluteY1(element));
+    subCanvasY2 = Math.max(subCanvasY2, getElementAbsoluteY2(element));
+  });
+
+  let targetCanvas = canvas;
+
+  if ( visibleOnly ) {
+    targetCanvas = document.createElement('canvas');
+    targetCanvas.style.display = 'none';
+    document.body.appendChild(targetCanvas);
+    targetCanvas.width = subCanvasX2 - subCanvasX1 + padding * 2;
+    targetCanvas.height = subCanvasY2 - subCanvasY1 + padding * 2;
+    const targetCanvas_ctx = targetCanvas.getContext('2d');
+
+    if ( background ) {
```

Comment:
We should figure out a way to get prettier running so that we don’t have style conflicts. Codesandbox runs it by default.